flake.nix

```diff
@@ -8,30 +8,33 @@
     "tweag-jupyter.cachix.org-1:UtNH4Zs6hVUFpFBTLaA4ejYavPo5EFFqgd7G7FxGW9g="
   ];
 
-  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
-  inputs.nixpkgs-stable.url = "github:nixos/nixpkgs/nixos-22.05";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  inputs.nixpkgs-stable.url = "github:nixos/nixpkgs/nixos-23.05";
   inputs.flake-compat.url = "github:edolstra/flake-compat";
   inputs.flake-compat.flake = false;
   inputs.flake-utils.url = "github:numtide/flake-utils";
   inputs.ihaskell.url = "github:ihaskell/ihaskell";
   inputs.ihaskell.inputs.nixpkgs.follows = "nixpkgs";
-  inputs.ihaskell.inputs.flake-compat.follows = "flake-compat";
+  inputs.ihaskell.inputs.flake-compat.follows = "";
   inputs.ihaskell.inputs.flake-utils.follows = "flake-utils";
   inputs.nix-dart.url = "github:djacu/nix-dart";
   inputs.nix-dart.inputs.nixpkgs.follows = "nixpkgs";
   inputs.nix-dart.inputs.flake-utils.follows = "flake-utils";
   inputs.npmlock2nix.url = "github:nix-community/npmlock2nix/0ba0746d62974403daf717cded3f24c617622bc7";
   inputs.npmlock2nix.flake = false;
-  inputs.opam-nix.url = "github:tweag/opam-nix/75199758e1954f78286e7e79c0e3916e28b732b0";
-  inputs.opam-nix.inputs.flake-compat.follows = "flake-compat";
+  inputs.opam-nix.url = "github:tweag/opam-nix";
+  inputs.opam-nix.inputs.flake-compat.follows = "";
   inputs.opam-nix.inputs.flake-utils.follows = "flake-utils";
   inputs.opam-nix.inputs.nixpkgs.follows = "nixpkgs";
   inputs.pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
   inputs.pre-commit-hooks.inputs.flake-utils.follows = "flake-utils";
   inputs.pre-commit-hooks.inputs.nixpkgs.follows = "nixpkgs";
+  inputs.pre-commit-hooks.inputs.flake-compat.follows = "";
+  # https://github.com/nix-community/poetry2nix/pull/1329
   inputs.poetry2nix.url = "github:nix-community/poetry2nix";
```

Comment:
Probably needs lock file update: 
>error: attribute 'overlays' missing
       at /nix/store/cygya77xpaz7zxkdmr9klsgxfc1hc60y-source/flake.nix:71:11:
           70|         pkgs = nixpkgs.legacyPackages.${system}.appendOverlays [
           71|           poetry2nix.overlays.default
             |           ^
           72|         ];
       Did you mean overlay?